pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -0,0 +1,97 @@
+package zero
+
+import (
+	"fmt"
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+	"math/big"
+)
+
+func newEcNegateHint(point hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcNegate",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+			//> y = pack(ids.point.y, PRIME) % SECP_P
+			//> # The modulo operation in python always returns a nonnegative number.
+			//> value = (-y) % SECP_P
+
+			secPBig, ok := utils.GetSecPBig()
+			if !ok {
+				return fmt.Errorf("GetSecPBig failed")
+			}
+
+			baseBig, ok := utils.GetEcBaseBig()
+			if !ok {
+				return fmt.Errorf("GetEcBaseBig failed")
+			}
+
+			pointValues, err := hinter.GetConsecutiveValues(vm, point, int16(6))
+			if err != nil {
+				return err
+			}
+
+			yD0, err := pointValues[3].FieldElement()
+			if err != nil {
+				return err
+			}
+			yD1, err := pointValues[4].FieldElement()
+			if err != nil {
+				return err
+			}
+			yD2, err := pointValues[5].FieldElement()
+			if err != nil {
+				return err
+			}
+
+			//> def pack(z, prime):
+			//>     """
+			//>     Takes an UnreducedBigInt3 struct which represents a triple of limbs (d0, d1, d2) of field
+			//>     elements and reconstructs the corresponding 256-bit integer (see split()).
+			//>     Note that the limbs do not have to be in the range [0, BASE).
+			//>     prime should be the Cairo field, and it is used to handle negative values of the limbs.
+			//>     """
+			//>     limbs = z.d0, z.d1, z.d2
+			//>     return sum(as_int(limb, prime) * (BASE**i) for i, limb in enumerate(limbs))
+
+			valueBig := big.NewInt(0)
+			primeBig := fp.Modulus()
+
+			for idx, yD := range []*fp.Element{yD0, yD1, yD2} {
+				var yDBig big.Int
+				yD.BigInt(&yDBig)
+
+				//> as_int(limb, prime)
+				if yDBig.Cmp(new(big.Int).Div(primeBig, big.NewInt(2))) != -1 {
+					yDBig.Sub(&yDBig, primeBig)
+				}
+
+				valueToAddBig := new(big.Int).Exp(baseBig, big.NewInt(int64(idx)), nil)
+				valueToAddBig.Mul(valueToAddBig, &yDBig)
+				valueBig.Add(valueBig, valueToAddBig)
+			}
```

Comment:
Idk if this is a valid case, but shouldn't the product of pack method be also modulo SECP_P?
`y = pack(ids.point.y, PRIME) % SECP_P`

Reply:
Nice catch! I missed this.